### ISSUE TITLE
resolve config path fix needed in patch release

### ISSUE THREAD

**Issue by brodybits** [apache/cordova-osx#52]
As pointed out by @erisu (Bryan Ellis) in [[1]](https://issues.apache.org/jira/browse/CB-14228?page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel&focusedCommentId=16578014#comment-16578014):

[apache/cordova-common#32 (apache/cordova-common GH-32)](https://github.com/apache/cordova-common/pull/32) - fix resolve config path on macOS ("osx") was merged on master branch of cordova-common, needed on `cordova@2.2.x` to be part of `cordova-osx@4.0.x`

[1] <https://issues.apache.org/jira/browse/CB-14228?page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel&focusedCommentId=16578014#comment-16578014>

**Comment by brodybits** [comment]
The open question from my side is whether or not this issue should block publishing of `4.0.2` patch release as made in GH-51 (PR #51). I would rather not block the `4.0.2` release since it already resolves `npm audit` issues and included some much needed updates to `cordova.js`.

**Comment by janpio** [comment]
I don't understand the issue.

**Comment by erisu** [comment]
@janpio if you mean the issue I fixed in my PR, basically if you install a plugin and build the app on the second or more times, the "Device is Ready" does not get triggered.

Without my fix, as a temporary workaround, the user will need to re-add the platform every-time before build so the issue does not happen.

*Sample use case displaying issue*
These steps I also outlined in the ticket before.
```
$ cordova create myapp org.apache.myapp myapp
$ cd myapp
$ cordova platform add osx
$ cordova plugin add cordova-plugin-file
$ cordova run osx
$ cordova run osx  // (broken, the device is ready is not triggered)
$ cordova run osx  // (broken, the device is ready is not triggered)
...
```
Same applies to the build step since it goes through the same process.

If we know that this issue still exists, maybe we can mention the `temporary workaround` solution in the release notes? Always remove and re-add the platform back?

```
$ cordova create myapp org.apache.myapp myapp
$ cd myapp
$ cordova platform add osx
$ cordova plugin add cordova-plugin-file
$ cordova run osx
$ cordova platform rm osx
$ cordova platform add osx
$ cordova run osx
$ cordova platform rm osx
$ cordova platform add osx
$ cordova run osx
...
```

**Comment by erisu** [comment]
There is one more alternative solution.
```
$rm -rf ./platforms/osx/build/<APP_NAME>.app/Contents/Resources/config.xml
```
By doing this, during the prepare step, it will recreate the `config.xml` file properly.

**Comment by janpio** [comment]
No, I don't understand what @brodybits wrote in https://github.com/apache/cordova-osx/issues/52#issue-350033413 The words and sentence structure don't make any sense to me.

**Comment by shazron** [comment]
The issue is now moot since we already published cordova-osx@4.0.2 already, and the new cordova-common with @erisu's fix will need to go in the next patch release. All we need to do is publish a new cordova-common, and cordova-osx depend on that new version.

**Comment by erisu** [comment]
Closing as users can upgrade to `cordova-osx@5.x` which contains the fix.
